near-contract-standards/src/non_fungible_token/enumeration_impl.rs

```diff
@@ -0,0 +1,99 @@
+use crate::non_fungible_token::core_impl::NonFungibleToken;
+use crate::non_fungible_token::enumeration::NonFungibleTokenEnumeration;
+use near_sdk::json_types::{U128, ValidAccountId};
+use crate::non_fungible_token::token::Token;
+use std::collections::HashMap;
+use near_sdk::AccountId;
+
+type TokenId = String;
+
+impl NonFungibleToken {
+    // Helper function used by a enumerations methods
+    // Note: this method is not exposed publicly to end users
+    pub fn enum_get_token(&self, owner_id: AccountId, token_id: TokenId) -> Token {
+        let metadata = self.token_metadata_by_id.as_ref().unwrap().get(&token_id);
+        let approved_account_ids = self
+            .approvals_by_id.as_ref()
+            .unwrap().get(&token_id).or_else(|| Some(HashMap::new()));
```

Comment:
```suggestion
        let approved_account_ids =
            Some(self.approvals_by_id.as_ref().unwrap().get(&token_id).unwrap_or_default());
```

Is this intentionally defaulting the hashmap instead of leaving empty? Why is this done with this and not metadata?

Reply:
Good question, would be curious to know more about this. Got me reading more about it here:
https://doc.rust-lang.org/src/std/collections/hash/map.rs.html#1123-1132
This commit suggestion got outdated due to some fmt corrections, but I've pushed it up.

Reply:
Well it would just change how this is handled externally, or how it's serialized. It's just the difference between `Some` empty map and `None`. I'll try to find out what the differences are in how it's handled

Reply:
Cool thanks! No sweat on digging deeper unless you feel like it; I was itching the scratch of curiosity. Good thoughts